app/src/main/java/ch/epfl/skysync/models/calendar/AvailabilityCalendar.kt

```diff
@@ -44,4 +52,45 @@ class AvailabilityCalendar : CalendarModel<Availability>() {
     }
     return nextAvailabilityStatus
   }
+
+  /**
+   * Returns the differences between this and the other calendar, from the point of view of the
+   * other calendar (that is, [CalendarDifferenceType.ADDED] means that [other] has an added
+   * availability compared to this)
+   *
+   * @param other The other calendar
+   */
+  fun getDifferencesWithOtherCalendar(
+      other: AvailabilityCalendar
+  ): List<Pair<CalendarDifferenceType, Availability>> {
+    val differences = mutableListOf<Pair<CalendarDifferenceType, Availability>>()
+    for (otherAvailability in other.cells) {
+      val date = otherAvailability.date
+      val timeSlot = otherAvailability.timeSlot
+      val thisAvailability = getByDate(date, timeSlot)
+      if (otherAvailability == thisAvailability) {
+        continue
+      }
+      if (thisAvailability == null || thisAvailability.status == AvailabilityStatus.UNDEFINED) {
```

Comment:
UNDEFINED is a psydo status for the availability i.e. it might be returned by a function to the UI components however it is never saved as a state inside the calendar

Reply:
I never make such assumptions unless explicitly stated, but we can state it as such.